platforms/documentation/docs/src/docs/userguide/dep-man/declaring-depedencies/dependency_constraints.adoc

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2023 Gradle, Inc.
+//
+// Licensed under the Creative Commons Attribution-Noncommercial-ShareAlike 4.0 International License.;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://creativecommons.org/licenses/by-nc-sa/4.0/
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+[[dependency-constraints]]
+= Declaring Dependency Constraints
+
+*Dependency constraints* function similarly to <<declaring_dependencies_basics.adoc#declaring-dependencies-basics,dependencies>>, with the key distinction that they do not introduce a dependency themselves.
+Instead, _constraints define version requirements_ that influence the resolution process when a dependency is brought into the project by other means.
+
+Although constraints are not <<rich_versions.adoc#sec:strict-version,strict versions>> by default, you can specify a strict version constraint if needed.
+Once the dependency is included, the version specified by the constraint participates in conflict resolution just as if it were declared as a direct dependency.
+
+When developing a single-project library, *constraints* can be directly declared alongside direct dependencies.
+However, when developing multi-project libraries and applications, dependencies are best declared centrally in a <<platforms.adoc#sec:sharing-dep-versions-between-projects,platform>>:
+
+====
+include::sample[dir="snippets/java-platform/recommender/kotlin/platform",files="build.gradle.kts[tags=full-platform]"]
+include::sample[dir="snippets/java-platform/recommender/groovy/platform",files="build.gradle[tags=full-platform]"]
+====
+
+In general, a dependency is considered either a **direct** dependency, or a **transitive** dependency:
+
+- *Direct dependencies* are _directly required by a component_.
+- *Transitive dependencies* are dependencies that a component needs, but only because another direct or transitive dependency needs them.
```

Comment:
❗ This is confusing.  Direct deps are directly specified within that component's build or metadata.  Transitive dependencies are NOT specified in this way, they are pulled in transitively as deps of the direct deps.

However, a component may require both direct AND TRANSITIVE dependencies to compile or run.  (If so, it should probably declare those deps DIRECTLY, but this is perhaps closer to a best practice than a requirement.)

Reply:
Great tip. Updated accordingly.